intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+#### Stationary distribution as an eigenvector
+
+If you’ve spend more time in linear algebra land, you’ll be pleased to hear that common way to compute this (in algebraic graph theory) is to convert the graph to a matrix _P_ where each entry <img src="https://render.githubusercontent.com/render/math?math=P_{uv}"> is the transition probability from node _u_ to node _v_. Each dimension corresponds to a node, and the matrix maps from all nodes to all nodes. Then finding the stationary distribution is the same as finding [the eigenvector](https://www.youtube.com/watch?v=PFDu9oVAE-g) of the transition matrix that corresponds to eigenvalue 1; by the definition of an eigenvector, this vector is the distribution of value to each node such that if value flowed down each edge in proportion to its weight, the resulting value distribution would remain the same.
+
+#### Incorporate edge weights
+
+We can adapt the algorithm to account for edge weights on the input graph fairly intuitively – we could add weights to the links by increasing the number of edges for some types of actions (e.g. 👍=1 edge, ♥️=2 edges). In the probability matrix formulation, it’s simple to increase the transition probability of one edge. Then, instead of assigning equal probability to all outbound edges, we can simply normalize the weights so that they sum to one and this provides us with transition probabilities needed for <img src="https://render.githubusercontent.com/render/math?math=P_{uv}">.
+
+#### Incorporate node weights
+
+There’s also a cleverly simple way to account for the existence of node weights. One requirement for the existence of a stationary distribution that was skipped is the concept of ergodicity, wherein every state must be reachable from any other state, given a long enough probabilistic walk across the graph’s edges†. To achieve ergodicity, SourceCred uses a "teleportation" system by adding a probability, ⍺, of teleporting to a random location in the graph. SourceCred currently implements this with a seed node, which all nodes teleporting to with likelihood ⍺. From the seed node, the probability of landing on a given node in the graph is then given by the node weights. If we use _A_ to describe the weights of each node in a vector form, then our old transition matrix <img src="https://render.githubusercontent.com/render/math?math=P_{uv}"> becomes the new transition matrix <img src="https://render.githubusercontent.com/render/math?math=(1-\alpha)P_{uv} + \alpha A">. The stationary distribution is then the vector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> such that <img src="https://render.githubusercontent.com/render/math?math=\left((1-\alpha)P_{uv} + \alpha A\right)\vec v = \vec v">.
+
```

Comment:
Did you mean "<img src="https://render.githubusercontent.com/render/math?math=(1-\alpha)P_{uv}"> + <img src="https://render.githubusercontent.com/render/math?math=\alpha B"> where each row of <img src="https://render.githubusercontent.com/render/math?math=B"> is <img src="https://render.githubusercontent.com/render/math?math=A^T">" instead of <img src="https://render.githubusercontent.com/render/math?math=(1-\alpha)P_{uv} + \alpha A">?

Reply:
I definitely wanted a new transition matrix, and I think you named exactly the matrix that I intended to describe.
However, it's also come to my attention that this isn't actually how it's implemented - rather we actually add a seed node to the matrix. We add a row of alphas to the bottom of the matrix, as all nodes have alpha probability of transitioning to it, and we append $A$ to the right side of the matrix, with a 0 in the bottom right corner.  This preserves the sparseness of the matrix, and makes computation simpler, and might also make time-based cred (epoch nodes and fibration stuff) easier as well.  
I'll fix this when I add more information on time-cred shortly and comment here after that.

Reply:
One thing that will be very useful to different kinds of learners is to (along with the equation, description, and any illustrations) there is a link to the algorithm in the code. I don't have a strong statistics background so I usually go backwards from code -> equations to understand something best.

Reply:
@vsoch, since I'm much less familiar with the code, I'd really appreciate it if you added links to the code.  Some links might have to wait until the Fibration refactor of time-weighted cred merges, but I'm strongly onboard with having those links in here.

Reply:
I'd definitely be happy to do this! I think the code is going to be changed a lot so we likely should hold off for now. I'll open an issue with detail about it.

Reply:
awesome, thank you!